CBLClient/Apps/CBLTestServer-Dotnet/TestServer/P2PMethods.cs

```diff
@@ -35,16 +42,106 @@ public class P2PMethods
+            URLEndpointListenerConfiguration urlEndpointListenerConfig = new URLEndpointListenerConfiguration(db);
+            string tlsAuthType = postBody["tls_auth_type"].ToString();
+            Boolean disableTls = Convert.ToBoolean(postBody[V].ToString());
+            Boolean tls_authenticator = Convert.ToBoolean(postBody["tls_authenticator"].ToString());
+
+
+            AddStatus("URL Start SERVER");
+            if (port > 0)
+            {
+                urlEndpointListenerConfig.Port = (ushort)port;
+            }
+
+            if (postBody.ContainsKey("basic_auth"))
+            {
+               urlEndpointListenerConfig.Authenticator = MemoryMap.Get<ListenerPasswordAuthenticator>(postBody["basic_auth"].ToString());
+            }
+            urlEndpointListenerConfig.DisableTLS = disableTls;
+            if (tlsAuthType == "self_signed")
+            {
+                Stream stream;
+                _store = new X509Store(StoreName.My);
+                byte[] cert = null;
+                TLSIdentity.DeleteIdentity(_store, ServerCertLabel, null);
+                string certLocation = TestServer.FilePathResolver("certs/certs.p12", false);
+                byte[] certsData = File.ReadAllBytes(certLocation);
+                TLSIdentity identity = TLSIdentity.ImportIdentity(_store, certsData, "123", ServerCertLabel, null);
+                urlEndpointListenerConfig.TlsIdentity = identity;
+                
+
+            } else if (tlsAuthType == "self_signed_create")
+            {
+                AddStatus("CreateIdentity");
+                _store = new X509Store(StoreName.My);
+                TLSIdentity.DeleteIdentity(_store, ServerCertLabel, null);
+                TLSIdentity identity = TLSIdentity.CreateIdentity(false,
+                    new Dictionary<string, string>() { { Certificate.CommonNameAttribute, ServerCertLabel } },
+                    null,
+                    _store,
+                    ServerCertLabel,
+                    null);
+
+                urlEndpointListenerConfig.TlsIdentity = identity;
+            }
+            if (tls_authenticator) {
+                _store = new X509Store(StoreName.My);
+                TLSIdentity.DeleteIdentity(_store, ServerCertLabel, null);
+                string certLocation = TestServer.FilePathResolver("certs/client-ca.der", false);
+                byte[] caData = File.ReadAllBytes(certLocation);
+
+                var rootCert = new X509Certificate2(caData);
+                var auth = new ListenerCertificateAuthenticator(new X509Certificate2Collection(rootCert));
+                urlEndpointListenerConfig.Authenticator = auth;
+            }
+
+            URLEndpointListener urlEndpointListener = new URLEndpointListener(urlEndpointListenerConfig);
+            urlEndpointListener.Start();
+            AddStatus("Started the url listener and waiting for connection..On the below port");
+            Console.WriteLine(urlEndpointListener.Port);

```

Comment:
there are several Console.WriteLine() calls been used, are you able to use logger in TestServer?